netmaster/k8snetwork/networkpolicy.go

```diff
@@ -135,204 +257,489 @@ func (k8sNet *k8sContext) deleteEpg(networkname, epgName, policyName string) err
+			//as pod add event
+			k8sNet.processPodAddEvent(pod)
+		}
+
+	}
+}
 
-	if err = k8sNet.createEpg(nwName, epgName, policyName); err != nil {
-		npLog.Errorf("failed to update EPG %s, %s", epgName, err)
-		return
+//Parse Pod Info from receive Pod events
+func parsePodInfo(pod *v1.Pod) npPodInfo {
+	var pInfo npPodInfo
+	pInfo.nameSpace = pod.ObjectMeta.Namespace
+	for key, val := range pod.ObjectMeta.Labels {
+		pInfo.labelSelectors =
+			append(pInfo.labelSelectors, (key + "=" + val))
 	}
+	pInfo.IP = pod.Status.PodIP
+	return pInfo
+}
 
-	if err = k8sNet.createRule(policyName, defaultRuleID, action); err != nil {
-		npLog.Errorf("failed to update default rule, %s", err)
-		return
+//Get  Network Policy object sets which ToSpec labelMap information match
+//with given pods labelMap
+func (k8sNet *k8sContext) getMatchToSpecPartNetPolicy(
+	podInfo npPodInfo) []*k8sNetworkPolicy {
+	var toPartPolicy []*k8sNetworkPolicy
+	nwPolicy, ok := k8sNet.nwPolicyPerNameSpace[podInfo.nameSpace]
+	if !ok {
+		npLog.Warnf("No NetworkPolicy for NameSpace:%v",
+			podInfo.nameSpace)
+		return nil
 	}
+	for _, nwPol := range nwPolicy {
+		for _, label := range podInfo.labelSelectors {
+			//Collect networkPolicy object which match with pods
+			//Labels
+			if _, ok := nwPol.PodSelector.labelPodMap[label]; ok {
+				toPartPolicy = append(toPartPolicy, nwPol)
+				//	npLog.Infof("policy :%+v", nwPol)
+				break
+			}
+		}
+	}
+	return toPartPolicy
 }
 
-func (k8sNet *k8sContext) deleteDefaultIngressPolicy(ns string) {
-	nwName := ns + "-" + defaultNetworkName
-	policyName := ns + "-" + defaultPolicyName
-	epgName := ns + "-" + defaultEpgName
+//Get Network Policy object sets which FromSpec, labelMap information match
+//with given pods labelMap
+func (k8sNet *k8sContext) getMatchFromSpecPartNetPolicy(
+	podInfo npPodInfo) []*k8sNetworkPolicy {
 
-	var err error
+	var fromPartPolicy []*k8sNetworkPolicy
+	//NetworkPolicy master object   on pods Namespace
+	nwPolicy, ok := k8sNet.nwPolicyPerNameSpace[podInfo.nameSpace]
+	if !ok {
+		npLog.Infof("Pod namespace doesn't have any policy config")
+		return nil
+	}
+	//Build list of networkPolicy object which fromSpec belongs to given
+	//pods Info
+	for _, l := range podInfo.labelSelectors {
+		for _, nwPol := range nwPolicy {
+			for _, ingress := range nwPol.Ingress {
+				//PodSelector on FromSpec part of policy Object
+				for _, podSelector := range ingress.
+					IngressPodSelector {
+					npLog.Infof("labelMap:%+v",
+						podSelector.labelPodMap)
+					if _, ok :=
+						podSelector.labelPodMap[l]; ok {
+						fromPartPolicy =
+							append(fromPartPolicy,
+								nwPol)
+						break
+					}
+				}
+			}
+		}
+	}
+	return fromPartPolicy
+}
 
-	if err = k8sNet.deleteRule(policyName, defaultRuleID); err != nil {
-		npLog.Errorf("failed to delete default rule, %s", err)
+//Consolidate all Ips belongs to Label for Pod Selector object
+func (k8sNet *k8sContext) updatePodSelectorPodIps(
```

Comment:
this function doesn't appear to use k8sContext, be more appropriate to be part of the k8sPodSelector struct?